components/sup/tests/utils/mod.rs

```diff
@@ -120,3 +125,46 @@ where
         }
     }
 }
+
+fn write_default_svc_user_and_group_metafiles<P>(hab_pkg_path: P)
+where
+    P: AsRef<Path>,
+{
+    let svc_user_metafile = hab_pkg_path.as_ref().join(MetaFile::SvcUser.to_string());
+    let svc_group_metafile = hab_pkg_path.as_ref().join(MetaFile::SvcGroup.to_string());
+
+    if !svc_user_metafile.is_file() {
+        write_metafile(
+            svc_user_metafile,
+            users::get_current_username()
+                .expect("Could not determine current username")
+                .as_str(),
+        );
+    }
+
+    if !svc_group_metafile.is_file() {
+        write_metafile(
+            svc_group_metafile,
+            users::get_current_groupname()
+                .expect("Could not determine current groupname")
+                .as_str(),
+        );
+    }
+}
+
+fn write_metafile<P>(metafile: P, content: &str)
```

Comment:
What do you think about taking a `ToString` here instead of a `&str`? 

One of the things I've been trying to do is make these testing functions as general as can be, thus the liberal use of things like `AsRef<Path>` and `ToString` in the type signatures. I was basically getting tired of functions that were too narrowly scoped, causing me to have to write unnecessarily noisy test code like `do_the_thing(String::from("blah"))` instead of just `do_the_thing("blah")`.

Reply:
I would have used `ToString`, but in this case we don't need the extract allocation of the file contents, only a borrow. I could update the function sig. to take a `Vec<u8>` which is what is being written, but I went with the easiest to get the write descriptor closed.